vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthWaitForEmailFragment.kt

```diff
@@ -38,45 +42,60 @@ data class FtueAuthWaitForEmailFragmentArgument(
 /**
  * In this screen, the user is asked to check his emails
  */
-class FtueAuthWaitForEmailFragment @Inject constructor() : AbstractFtueAuthFragment<FragmentLoginWaitForEmailBinding>() {
+class FtueAuthWaitForEmailFragment @Inject constructor(
+        private val themeProvider: ThemeProvider
+) : AbstractFtueAuthFragment<FragmentFtueWaitForEmailVerificationBinding>() {
 
     private val params: FtueAuthWaitForEmailFragmentArgument by args()
+    private var inferHasLeftAndReturnedToScreen = false
 
-    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentLoginWaitForEmailBinding {
-        return FragmentLoginWaitForEmailBinding.inflate(inflater, container, false)
+    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentFtueWaitForEmailVerificationBinding {
+        return FragmentFtueWaitForEmailVerificationBinding.inflate(inflater, container, false)
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-
         setupUi()
     }
 
+    private fun setupUi() {
+        views.emailVerificationGradientContainer.setBackgroundResource(
+                when (themeProvider.isLightTheme()) {
+                    true  -> R.drawable.bg_waiting_for_email_verification
+                    false -> R.drawable.bg_color_background
+                }
+        )
+        views.emailVerificationTitle.text = R.string.ftue_auth_email_verification_title.colorTerminatingFullStop(R.attr.colorSecondary)
+        views.emailVerificationSubtitle.text = getString(R.string.ftue_auth_email_verification_subtitle, params.email)
+        views.emailVerificationResendEmail.debouncedClicks {
+            viewModel.handle(OnboardingAction.PostRegisterAction(RegisterAction.SendAgainThreePid))
+        }
+    }
+
+    private fun Int.colorTerminatingFullStop(@AttrRes color: Int): CharSequence {
+        return getString(this).colorTerminatingFullStop(ThemeUtils.getColor(requireContext(), color))
+    }
+
     override fun onResume() {
         super.onResume()
-
+        showLoadingIfReturningToScreen()
         viewModel.handle(OnboardingAction.PostRegisterAction(RegisterAction.CheckIfEmailHasBeenValidated(0)))
     }
 
+    private fun showLoadingIfReturningToScreen() {
+        when (inferHasLeftAndReturnedToScreen) {
+            true  -> views.emailVerificationWaiting.isVisible = true
```

Comment:
Why are we not always showing the loader by the way?

Reply:
the idea was to avoid the impression that app was busy/loading when we we're waiting for the user to click the link in the email verification

the middle ground was to show the spinner if we detect the user has left and returned to app as we're inferring that they've clicked the link and the app is now waiting for the homeserver to respond